From c9ec3d35bae3220b5b79406480beb4e01b601681 Mon Sep 17 00:00:00 2001
From: Michael-CY Lee <michael-cy.lee@mediatek.com>
Date: Thu, 12 Jun 2025 10:03:09 +0800
Subject: [PATCH 100/128] mtk: mac80211: do not skip non-TX BSS in per-STA
 profile

To be honest, I did not fully understand the comment in the code
that explains the reason for skipping non-TX BSS in the per-STA
profile. It seems like the concern is about address calculation
in MBSSID, but the address of the non-TX BSS is provided in the
STA Info field in the per-STA profile.

Moreover, skipping non-TX BSS in the per-STA profile prevents
the STA from connecting to all the AP MLD's links.

Signed-off-by: Michael-CY Lee <michael-cy.lee@mediatek.com>
---
 net/wireless/scan.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/net/wireless/scan.c b/net/wireless/scan.c
index 3a2a29a2..b36c38b7 100644
--- a/net/wireless/scan.c
+++ b/net/wireless/scan.c
@@ -3067,8 +3067,17 @@ cfg80211_parse_ml_elem_sta_data(struct wiphy *wiphy,
 		 * Without both pieces of information we cannot calculate the
 		 * reference BSSID, so simply ignore the BSS.
 		 */
+
+		/*
+		 * NonTxBSSID can be retrieved from the STA Info field in the
+		 * per-sta profile.
+		 * The BSSID can be directly retrieved without calculation of
+		 * address extension.
 		if (non_tx)
 			continue;
+		 */
+
+
 
 		/* We could sanity check the BSSID is included */
 
-- 
2.45.2

